syntax = "proto3";

package trace;

enum BranchType {
    JUMP = 0;
    CONDJUMP = 1;
    CALL = 2;
    INDJUMP = 3;
    INDCALL = 4;
}

message BranchEvent {
    uint64 source = 1;
    uint64 destination = 2;
    BranchType type = 3;
}

message MappingEvent {
    uint64 start = 1;
    uint64 size = 2;
    uint64 offset = 3;
    string filename = 4;
}

message Trace {
    repeated MappingEvent mappings = 1;
    repeated BranchEvent branches = 2;
}
